pkg/agent/datamodel/types.go

```diff
@@ -1031,6 +1031,11 @@ func (a *AgentPoolProfile) GetKubernetesLabels(rg string, deprecated bool, nvidi
 	if osSku != "" {
 		buf.WriteString(fmt.Sprintf(",kubernetes.azure.com/os-sku=%s", osSku))
 	}
+
+	if a.Distro == AKSUbuntuContainerd2004CVMGen2 {
```

Comment:
nit: let's go with ConfidentialVM and TrustedLaunch for the values here?

Reply:
assuming you mean kubernetes.azure.com/security-type=ConfidentialVM and not for the distro?